lib/rules/no-jquery.js

```diff
@@ -27,62 +28,93 @@ module.exports = {
   ERROR_MESSAGE,
 
   create(context) {
-    let destructuredAssignment;
-    let emberImportAliasName;
-    let jqueryImportAliasName;
     const report = function (node) {
       context.report(node, ERROR_MESSAGE);
     };
 
+    /**
+     * Reports each reference to the variable
+     *
+     * @param  {Variable} scopeVariable variable to report references for
+     * @return {number}                 the number of references to the variable
+     */
+    function reportVariableReferences(scopeVariable) {
+      // report each reference to the variable
+      scopeVariable.references.forEach((reference) => {
+        report(reference.identifier.parent);
+      });
+
+      return scopeVariable.references.length;
+    }
+
     return {
-      ImportDeclaration(node) {
-        emberImportAliasName = ember.getEmberImportAliasName(node);
+      /**
+       * Report all references whe importing jQuery
+       *
+       * eg; `import $ from 'jquery'` or `import jq from 'jquery'`
+       */
+      'ImportDeclaration[source.value="jquery"] ImportDefaultSpecifier'(node) {
+        const scopeVariable = findVariable(context.getScope(), node.local);
+        const hasAnyReferences = reportVariableReferences(scopeVariable);
 
-        if (!jqueryImportAliasName && node.source && node.source.value === 'jquery') {
-          jqueryImportAliasName = node.specifiers[0].local.name;
+        if (!hasAnyReferences) {
+          report(node.parent);
         }
       },
 
-      VariableDeclarator(node) {
-        if (emberImportAliasName) {
-          destructuredAssignment =
-            node.init && types.isMemberExpression(node.init)
-              ? node.id.name
-              : // assignment of type const $ = Ember.$;
-                utils
-                  .collectObjectPatternBindings(node, {
-                    [emberImportAliasName]: ['$'],
-                  })
-                  .pop();
+      /**
+       * Report all references when destructuring `Ember.$`
+       *
+       * eg; `const { $ } = Ember` or `const { $: jq } = Ember'`
```

Comment:
I would prefer to maintain the existing behavior of the rule of only flagging the function calls and not the destructuring statements since this ends up being a duplicate violation.